src/cloud_client/mod.rs

```diff
@@ -0,0 +1,11 @@
+use crate::errors::AppError;
+use std::path::PathBuf;
+
+pub mod dropbox;
+
+pub trait CloudClient {
+    fn download(&self, from_path: PathBuf, to_path: PathBuf) -> Result<(), AppError>;
+    fn upload(&self, from_path: PathBuf, to_path: PathBuf) -> Result<(), AppError>;
+    fn delete(&self, path: PathBuf) -> Result<(), AppError>;
+    fn list_entries(&self, path: PathBuf) -> Result<Vec<String>, AppError>;
+}
```

Comment:
I think there is no reason to use `PathBuf` for data that will be used in API only (`to_path`, `path`). As `PathBuf` can be non valid UTF-8 value and it will fail during serialization.